desktop/core/src/desktop/lib/botserver/views_tests.py

```diff
@@ -47,4 +52,57 @@ def test_say_hi_user(self):
       }
       response = say_hi_user("channel", "user_id")
       api_call.assert_called_with(api_method='chat.postMessage', json={'channel': 'channel', 'text': 'Hi <@user_id> :wave:'})
-      assert_true(response['ok'])
\ No newline at end of file
+      assert_true(response['ok'])
+  
+  def test_handle_on_message(self):
+    with patch('desktop.lib.botserver.views.say_hi_user') as say_hi_user:
+      
+      response = handle_on_message("channel", "bot_id", "text", "user_id")
+      assert_equal(response.status_code, 200)
+      assert_false(say_hi_user.called)
+      
+      handle_on_message("channel", None, None, "user_id")
+      assert_false(say_hi_user.called)
+
+      handle_on_message("channel", None, "text", "user_id")
+      assert_false(say_hi_user.called)
+
+      handle_on_message("channel", None, "hello hue test", "user_id")
+      assert_true(say_hi_user.called)
+
+  def test_handle_on_link_shared(self):
+    with patch('desktop.lib.botserver.views.slack_client.chat_unfurl') as chat_unfurl:
+      with patch('desktop.lib.botserver.views._make_unfurl_payload') as mock_unfurl_payload:
+        with patch('desktop.lib.botserver.views.Document2.objects.get') as document2_objects_get:
+          with patch('desktop.lib.botserver.views._get_gist_document') as _get_gist_document:
+        
+            client = make_logged_in_client(username="test", groupname="default", recreate=True, is_superuser=False)
+            user = User.objects.get(username="test")
+            channel_id = "channel_id"
+            message_ts = "123.1"
+
+            # qhistory link
+            links = [{"url": "https://demo.gethue.com/hue/editor?editor=123456"}]
+            doc_data = {
+              "dialect": "mysql",
+              "snippets": [{
+                "database": "hue",
+                "statement_raw": "SELECT 5000",
+              }]
+            }
+            document2_objects_get.return_value = Mock(data=json.dumps(doc_data), owner=user)
+            handle_on_link_shared(channel_id, message_ts, links)
```

Comment:
nit: usually new line before/after to separate:
- test prep
- call
- test check

same for gist link

Reply:
Fixed